dbms/src/Common/HashTable/ClearableHashMap.h

```diff
@@ -15,10 +15,10 @@ struct ClearableHashMapCell : public ClearableHashTableCell<Key, HashMapCell<Key
 };
 
 template<typename Key, typename Mapped, typename Hash>
-auto lookupResultGetKey(ClearableHashMapCell<Key, Mapped, Hash> * cell) { return &cell->getFirst(); }
+ALWAYS_INLINE inline auto lookupResultGetKey(ClearableHashMapCell<Key, Mapped, Hash> * cell) { return &cell->getFirst(); }
```

Comment:
template functions don't need `inline`

Reply:
Without `inline`, gcc-9 gives me warnings like these:

```
/home/akuzm/ch4/ch/dbms/src/Common/HashTable/HashTable.h:211:20: warning: always_inline function might not be inlinable [-Wattributes]
  211 | ALWAYS_INLINE auto lookupResultGetKey(HashTableCell<Key, Hash, State> * cell)
      |                    ^~~~~~~~~~~~~~~~~~
```

Not sure if it's a broken warning, or there is some special rule for partial specialization.

Reply:
yeah, specializations need inline. Not sure about partial ones

Reply:
But templates and partial templates should be exempt from ODR, which means `inline` is not needed.